### PR TITLE
DPTString: replace invalid characters with question marks

### PR DESCRIPTION
**Pull request by farmio**
## Description

DPTString: replace invalid characters with question marks

When trying to send an invalid character (eg. "š") we would raise an error at https://github.com/XKNX/xknx/blob/c19ea3d4347513d6052f358ea98948e248a68e97/xknx/telegram/apci.py#L271 because
```python
>>> ord("š")
353
>>> bytes((353,))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: bytes must be in range(0, 256)
```
and this would crash our outgoing TelegramQueue.

see https://community.home-assistant.io/t/outgoing-knx-telegrams-are-not-send-after-ha-core-update/340444

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] The documentation has been adjusted accordingly
- [x] The changes generate no new warnings
- [x] Tests have been added that prove the fix is effective or that the feature works
- [x] The changes are documented in the changelog
- [ ] The Homeassistant plugin has been adjusted in case of new config options

### CODE REVIEW

**Comment by farmio** [comment]
oh come on @codecov you are pulling a coveralls now!

File: xknx/dpt/dpt_string.py [hunk 1]
```diff
@@ -36,11 +36,11 @@ def to_knx(cls, value: str) -> tuple[int, ...]:
...
                 raise ValueError
-            raw = []
-            for character in knx_value:
-                raw.append(ord(character))
+            raw = [ord(character) for character in knx_value]
             raw.extend([0] * (cls.payload_length - len(raw)))
-            return tuple(raw)
+            # replace invalid characters with question marks
+            # bytes(knx_value, 'ascii') would raise UnicodeEncodeError
+            return tuple(map(lambda char: char if char <= 0xFF else ord("?"), raw))
```
Comment (marvin-w) [inline xknx/dpt/dpt_string.py:43]: looks very cryptic on first sight, but there is a comment explaining it so I guess its fine

**Review by marvin-w — APPROVED** [review]
LGTM

**Comment by marvin-w** [comment]
Can be merged after CI is done, conflicts are solved.

**Pull request merged**